clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -142,9 +139,7 @@ private NetworkClient(MetadataUpdater metadataUpdater,
         this.correlation = 0;
         this.randOffset = new Random();
         this.requestTimeoutMs = requestTimeoutMs;
-        this.nodesEverSeen = new ArrayList<>();
-        this.nodesEverSeenById = new HashMap<>();
-        
+        this.bootstrapNodes = this.metadataUpdater.fetchNodes();
```

Comment:
What we are getting here is not necessarily the bootstrap nodes that the user has passed as it depends on what the caller passes (the current implementation does that, but it could easily be changed). 3 options I can think of:
* Leave the current implementation, but change the name of the field to something like `initialNodes`
* Add an additional constructor parameter called `bootstrapNodes`
* Leave as is